src/coreclr/jit/compiler.cpp

```diff
@@ -4737,6 +4737,13 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     fgStress64RsltMul();
 #endif // DEBUG
 
+    if (opts.OptimizationEnabled())
+    {
+        // Build post-order that morph will use, and remove dead blocks
+        //
+        DoPhase(this, PHASE_DFS_BLOCKS, &Compiler::fgDfsBlocksAndRemove);
```

Comment:
I think we can move this much earlier and benefit from it in local morph, early liveness and physical promotion too. However, physical promotion may create a scratch BB so we should do that as a separate canonicalization step before the DFS. I'll leave it for future work, right now I'm just interested in moving towards deleting the old DFS.

Reply:
Perhaps we should just always create a scratch BB early on and remove all the special checks later where we decide to add one if we need it.

Reply:
Yeah, I think that'd be good.